central/sensor/service/connection/manager_impl.go

```diff
@@ -306,6 +311,38 @@ func (m *manager) HandleConnection(ctx context.Context, sensorHello *central.Sen
 	return err
 }
 
+func (m *manager) handleConnectionError(clusterID string, err error) {
+	if e, ok := status.FromError(err); ok {
+		if e.Code() == codes.ResourceExhausted {
+			log.Warnf("gRPC connection received a payload too large to be processed: %s", err)
+			log.Warnf("Increase ROX_GRPC_MAX_MESSAGE_SIZE to allow larger payloads. Central will temporarily disable client reconciliation to avoid receiving large payloads from Sensor.")
+			pref := m.GetConnectionPreference(clusterID)
+			pref.SendDeduperState = false
+			m.connectionPreference.Store(clusterID, pref)
+		}
+	}
+	log.Warnf("Connection to server in cluster %s terminated: %v", clusterID, err)
+}
+
+// GetConnectionPreference returns the connection preference for cluster ID.
+func (m *manager) GetConnectionPreference(clusterID string) Preferences {
+	var p Preferences
+	pref, exists := m.connectionPreference.Load(clusterID)
+	if !exists {
+		p = Preferences{
+			SendDeduperState: true,
+		}
+		m.connectionPreference.Store(clusterID, p)
+	} else {
+		var ok bool
+		p, ok = (pref).(Preferences)
+		if !ok {
+			log.Warnf("Incorrect entry in Connection preferences map for cluster ID: %s", clusterID)
+		}
+	}
```

Comment:
Wouldn't that work? That should set the `p` right?
```suggestion
	} else if p, ok := (pref).(Preferences); !ok {
		log.Warnf("Incorrect entry in Connection preferences map for cluster ID: %s", clusterID)
	}
```

Reply:
I think you would have to define `ok` out of the `if else` and change the line to:

```golang
} else if p, ok = (pref).(Preferences); !ok {
```

Otherwise I think the code is defining a new variable `p` in the scope of that `else if` and not setting the `p` outside.

Reply:
Done